lib/datadog/appsec/event.rb

```diff
@@ -48,6 +48,8 @@ def self.record(span, *events)
       end
 
       def self.record_via_span(span, *events)
+        return unless span
```

Comment:
I'd do that at or around the call site.

I mean `record_via_span` doesn't make sense to be called without a span, which interestingly pointed me to the more pressing issue: the rate limiter will still rate limit even though it doesn't record!

Reply:
I also think we should address that using the `contrib/*/integration_spec.rb` shared examples:

I prepped some `let` here:

https://github.com/DataDog/dd-trace-rb/blob/78ca839b502d4a467c97dbf95b58f0707773a07c/spec/datadog/appsec/contrib/rack/integration_test_spec.rb#L22

... and had a TODO here:

https://github.com/DataDog/dd-trace-rb/blob/78ca839b502d4a467c97dbf95b58f0707773a07c/spec/datadog/appsec/contrib/rack/integration_test_spec.rb#L148

... for it to be implemented as an additional nested context like that one, except it's not appsec disabled but tracing:

https://github.com/DataDog/dd-trace-rb/blob/78ca839b502d4a467c97dbf95b58f0707773a07c/spec/datadog/appsec/contrib/support/integration/shared_examples.rb#L26